src/three-components/TextureUtils.ts

```diff
@@ -215,32 +201,47 @@ export default class TextureUtils extends EventDispatcher {
       environmentMap = await environmentMapLoads;
       skybox = await skyboxLoads;
 
-      // If environment is still null, then no specific environment URL was
-      // specified
-      if (environmentMap != null) {
-        environmentMap = environmentMap.texture
-      } else {
+      if (!!skyboxUrl) {
+        if (targetCache.has(skyboxUrl)) {
+          skybox = targetCache.get(skyboxUrl)!;
+        } else {
+          targetCache.set(skyboxUrl, skybox!);
+        }
+      }
+
+      if (environmentMapUrl == null) {
         if (skybox != null) {
           // Infer the environment from the skybox if we have one:
-          environmentMap = skybox.texture;
+          environmentMapUrl = skyboxUrl + ':env';
+          environmentMap = skybox;
         } else {
-          // Otherwise, no skybox URL was specified, so fall back to generating
-          // the environment:
-          // TODO(#336): can cache this per renderer and color
-          environmentMap = this.environmentMapGenerator.generate();
-          environmentMapWasGenerated = true;
+          // Otherwise, no skybox URL was specified, so fall back to
+          // generating the environment:
+          environmentMapUrl = 'generated:env';
+          if (!targetCache.has(environmentMapUrl)) {
+            this.environmentMapGenerator =
+                new EnvironmentMapGenerator(this.renderer);
+            environmentMap = this.environmentMapGenerator.generate();
+            environmentMapWasGenerated = true;
+          }
         }
+      } else {
+        environmentMapUrl = environmentMapUrl + ':env';
       }
 
       // Apply the PMREM pass to the environment, which produces a distinct
       // texture from the source:
-      const nonPmremEnvironmentMap = environmentMap;
-
-      environmentMap = this.pmremPass(nonPmremEnvironmentMap);
+      if (targetCache.has(environmentMapUrl)) {
+        environmentMap = targetCache.get(environmentMapUrl)!;
+      } else {
+        const nonPmremEnvironmentMap = environmentMap!.texture;
+        environmentMap = this.pmremPass(nonPmremEnvironmentMap);
+        targetCache.set(environmentMapUrl, environmentMap);
```

Comment:
This dualistic quality where the environment map may be PMREM'd or may not be adds an interesting complication. You would need to amend the skybox example I commented about above to do something like this when populating the cache in the environment map case:

```typescript
environmentCache.set(environmentUrl, this.loadEquirectAsCubeMap(
        environmentUrl,
        progressTracker ? progressTracker.beginActivity() : () => {})
    // Chain the promise so that the cached async work includes the PMREM pass:
    .then((environmentMap: WebGLRenderTarget) => this.pmremPass(environmentMap.texture)));
```

That would actually end up looking a lot cleaner than what we had before this PR was proposed...